ctl/vr/inductive_complex.h

```diff
@@ -0,0 +1,142 @@
+// of all the neighbors that have a lower ordering.
+template<typename Graph, typename Vertex, typename Points>
+void get_lower_neighbors (const Graph& g,
+                          const Vertex& v,
+                          Points& lower_neighbors) {
+    typedef typename boost::graph_traits<Graph>::adjacency_iterator Iterator;
+    Iterator ai, ai_end;
+    for( boost::tie(ai, ai_end) = boost::adjacent_vertices(v, g);
+         ai != ai_end; ++ai) {
+        if (*ai < v) { lower_neighbors.push_back( *ai); }
+    }
+} 
+
+// We add the given graph to the complex. Also, we update the cells vectors to
+// keep track of the dimensions of the cells.
+template<typename Graph, typename Complex, typename Vector>
+void add_graph_to_complex (const Graph& g,
+			   Complex& complex,
+			   Vector& k_cells,
+			   Vector& k_plus_one_cells) {
+    typedef typename Complex::Cell Simplex;
+    typedef typename boost::graph_traits< Graph>::vertex_iterator vertex_iterator;
+    typedef typename boost::graph_traits< Graph>::adjacency_iterator Iterator;
+    vertex_iterator vi, vlast;
+    for ( std::tie( vi, vlast) = boost::vertices( g); vi != vlast; ++vi) {
+        // adding {*vi} to both k_cells and complex
+        Simplex tau(1, *vi);
+        k_cells.push_back(tau);
+        complex.insert_open_cell(tau);
+        Iterator ai, ai_end;
+        for ( boost::tie(ai, ai_end) = boost::adjacent_vertices(*vi, g);
+              ai != ai_end; ++ai) {
+                // adding {*vi, *ai} to both k_plus_one_cells and complex
+                Simplex sigma( tau);
+                sigma.insert( *ai);
+                k_plus_one_cells.push_back(sigma);
+                complex.insert_open_cell(sigma);
+        }
+    }
+}
+
+// We loop through all nodes, selecting each one as the starting point for
+// constructing our simplicial complex.
+template<typename Graph, typename Complex> 
+void inductive_vr (const Graph& g, Complex& complex, std::size_t dimension) { 
+    typedef typename Complex::Cell Simplex;
+    typedef typename Simplex::value_type Vertex;
+    typedef typename Simplex::const_iterator simplex_iterator;
+    std::vector<Simplex> k_cells;
+    std::vector<Simplex> k_plus_one_cells;
+    // adding the graph to the complex
+    add_graph_to_complex (g, complex, k_cells, k_plus_one_cells);
+    // inductive algorithm
+    for ( int k = 1; k < dimension; k++) {
+        k_cells.swap(k_plus_one_cells);
+	k_plus_one_cells.clear();
+	for ( int i = 0; i < k_cells.size(); i++) {
+	    Simplex tau(k_cells[i]);
+	    // getting the intersection of all lower neighbors
+	    std::vector< Vertex> final_neighbors;
+            std::vector< Vertex> lower_neighbors;
+	    get_lower_neighbors(g, *(tau.begin()), final_neighbors);
+	    for(simplex_iterator vi = tau.begin(); vi != tau.end(); ++vi) {
+		lower_neighbors.clear();
+		get_lower_neighbors(g, *vi, lower_neighbors);
+		std::set< Vertex> finalN(final_neighbors.begin(), final_neighbors.end());
```

Comment:
Much of this is highly inefficient. A number of temporaries can be removed. Until this is resolved I won't accept the pull.

Also, I am adding the gtest unit test framework. Doing some unit testing on these algorithms would be useful.